### PR TITLE
[office-js-preview] (Outlook) Add prependOnSendAsync methods

### PR DESCRIPTION
**Pull request by samantharamon**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <None>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/office-js-preview/index.d.ts [hunk 1]
```diff
@@ -12370,6 +12370,89 @@ declare namespace Office {
...
+         * 
+         * - In a {@link https://learn.microsoft.com/office/dev/add-ins/outlook/smart-alerts-onmessagesend-walkthrough | Smart Alerts add-in},
+         * the prepend-on-send feature runs first.
+         * 
+         * - A new line is added after the prepended content.
+         * 
+         * - If multiple active add-ins call `prependOnSendAsync`, the order of the inserted content depends on the order in which the add-in runs.
+         * The content of the last run add-in appears above previously prepended content.
+         * 
+         * - If the add-in attempts to insert HTML into a plain text body, the content will not be prepended. Conversely, plain text will be inserted into an HTML body.
```
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts:12398]:
```suggestion
         * - If the add-in attempts to insert HTML into a plain text body, the content won't be prepended. Conversely, plain text will be inserted into an HTML body.
```

File: types/office-js-preview/index.d.ts [hunk 2]
```diff
@@ -12370,6 +12370,89 @@ declare namespace Office {
...
+         * see {@link https://learn.microsoft.com/office/dev/add-ins/outlook/understanding-outlook-add-in-permissions | Understanding Outlook add-in permissions}.
+         * 
+         * @remarks
+         * [Api set: Mailbox preview]
+         * 
+         * **{@link https://learn.microsoft.com/office/dev/add-ins/outlook/understanding-outlook-add-in-permissions | Minimum permission level}**: **read/write item**
+         *
+         * **{@link https://learn.microsoft.com/office/dev/add-ins/outlook/outlook-add-ins-overview#extension-points | Applicable Outlook mode}**: Compose
+         * 
+         * **Recommended**: Call `getTypeAsync` then pass its returned value to the `options.coercionType` parameter.
```
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts:12386]:
```suggestion
         * **Recommended**: Call `getTypeAsync`, then pass its returned value to the `options.coercionType` parameter.
```

File: types/office-js-preview/index.d.ts [hunk 3]
```diff
@@ -12370,6 +12370,89 @@ declare namespace Office {
...
+         * - If the add-in attempts to insert HTML into a plain text body, the content will not be prepended. Conversely, plain text will be inserted into an HTML body.
+         * 
+         * **Errors**:
+         * 
+         * - `DataExceedsMaximumSize`: The `data` parameter exceeds 5,000 characters.
+         * 
+         * - `InvalidFormatError`: The `options.coercionType` parameter is set to `Office.CoercionType.Html`, but the item body is in plain text format.
+         * 
+         * @param data - The string to be prepended to the beginning of the message or appointment body. The string is limited to 5,000 characters.
+         * @param options - An object literal that contains one or more of the following properties:-
```
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts:12407]: Errant hyphen?
```suggestion
         * @param options - An object literal that contains one or more of the following properties:
```

File: types/office-js-preview/index.d.ts [hunk 4]
```diff
@@ -12370,6 +12370,89 @@ declare namespace Office {
...
+         * see {@link https://learn.microsoft.com/office/dev/add-ins/outlook/understanding-outlook-add-in-permissions | Understanding Outlook add-in permissions}.
+         * 
+         * @remarks
+         * [Api set: Mailbox preview]
+         * 
+         * **{@link https://learn.microsoft.com/office/dev/add-ins/outlook/understanding-outlook-add-in-permissions | Minimum permission level}**: **read/write item**
+         *
+         * **{@link https://learn.microsoft.com/office/dev/add-ins/outlook/outlook-add-ins-overview#extension-points | Applicable Outlook mode}**: Compose
+         * 
+         * **Recommended**: Call `getTypeAsync` then pass its returned value to the `options.coercionType` parameter.
```
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts:12429]:
```suggestion
         * **Recommended**: Call `getTypeAsync`, then pass its returned value to the `options.coercionType` parameter.
```

File: types/office-js-preview/index.d.ts [hunk 5]
```diff
@@ -12370,6 +12370,89 @@ declare namespace Office {
...
+         * 
+         * - In a {@link https://learn.microsoft.com/office/dev/add-ins/outlook/smart-alerts-onmessagesend-walkthrough | Smart Alerts add-in},
+         * the prepend-on-send feature runs first.
+         * 
+         * - A new line is added after the prepended content.
+         * 
+         * - If multiple active add-ins call `prependOnSendAsync`, the order of the inserted content depends on the order in which the add-in runs.
+         * The content of the last run add-in appears above previously prepended content.
+         * 
+         * - If the add-in attempts to insert HTML into a plain text body, the content will not be prepended. Conversely, plain text will be inserted into an HTML body.
```
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts:12441]:
```suggestion
         * - If the add-in attempts to insert HTML into a plain text body, the content won't be prepended. Conversely, plain text will be inserted into an HTML body.
```

**Review by AlexJerabek — APPROVED** [review]
Looks good. Just a couple suggestions to take or leave.

File: types/office-js-preview/index.d.ts (hunk 3 above)
Comment (samantharamon) [reply types/office-js-preview/index.d.ts:12407]: We've had to use this format to list parameter properties. I'll follow up on this with Elizabeth when she returns from holiday.

**Pull request merged**